plugins/view-resources/src/components/ViewOptionsButton.svelte

```diff
@@ -53,9 +53,16 @@
       descriptor: viewlet.descriptor,
       variant: viewlet.variant ? viewlet.variant : { $exists: false }
     })
-
+    const hierarchy = client.getHierarchy()
```

Comment:
It is bad way to have hierarchy on top of component, since client could reconnect and refresh model and hierarchy will be old.

Reply:
Okay, will move it into mapping then. Should be better there, I assume.